### PR TITLE
Feature/small optimize

### PR DESCRIPTION
**Pull request by alexcos20**
- Create four separate threads, instead of one
- Fix queue processor

### CODE REVIEW

File: aquarius/events/events_monitor.py [hunk 1]
```diff
@@ -119,60 +118,83 @@ def __init__(self, web3, config_file):
...
 
-    def start_events_monitor(self):
-        if self._monitor_is_on:
-            return
+    def stop_monitor(self):
+        """Stops all threads for processing more data"""
+        self._thread_process_blocks_is_on = False
+        self._thread_process_queue_is_on = False
+        self._thread_process_ve_allocate_is_on = False
+        self._thread_process_purgatory_is_on = False
```
Comment (juanpablovs) [inline aquarius/events/events_monitor.py:135]: may be I am missing something but shouldn't it include self._monitor_is_on = False?

File: aquarius/events/events_monitor.py [hunk 2]
```diff
@@ -119,60 +118,83 @@ def __init__(self, web3, config_file):
...
+            self._thread_process_queue_is_on = True
+            t.start()
+        if self.ve_allocate:
+            t = Thread(target=self.thread_process_ve_allocate, daemon=True)
+            self._thread_process_ve_allocate_is_on = True
+            t.start()
+        if self.purgatory:
+            t = Thread(target=self.thread_process_purgatory, daemon=True)
+            self._thread_process_purgatory_is_on = True
+            t.start()
```
Comment (juanpablovs) [inline aquarius/events/events_monitor.py:154]: self._monitor_is_on = True?

**Review by juanpablovs — APPROVED** [review]
the while True loops consume a lot of resources and can have unexpected behavior, but I am not sure if using threading and making the calls asynchronous would also generate unexpected behavior due to thread separation

File: aquarius/events/events_monitor.py (hunk 1 above)
Comment (alexcos20) [reply aquarius/events/events_monitor.py:135]: no, I have removed _monitor_is_on and replaced it with 4 variables, one for each thread

File: aquarius/events/events_monitor.py (hunk 2 above)
Comment (alexcos20) [reply aquarius/events/events_monitor.py:154]: no, I have removed _monitor_is_on and replaced it with 4 variables, one for each thread

**Comment by alexcos20** [comment]
Idea is to have 4 threads, separated, so if one fails, it will not block everything.

while True: sleep() is cheap, it does not consume CPU cycles

**Review by calina-c — APPROVED** [review]

**Pull request merged**